### PR TITLE
Send data plane status update request

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

- Send data plane status update request
- Update process monitor to only send updates if the process have changed 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: internal/plugin/grpc_client_plugin.go [hunk 1]
```diff
@@ -124,10 +126,24 @@ func (gc *GrpcClient) Info() *bus.Info {
...
 }
 
-func (gc *GrpcClient) Process(_ context.Context, _ *bus.Message) {}
+func (gc *GrpcClient) Process(ctx context.Context, msg *bus.Message) {
+	switch {
+	case msg.Topic == bus.InstancesTopic:
+		if newInstances, ok := msg.Data.([]*v1.Instance); ok {
+			err := gc.sendDataPlaneStatusUpdate(ctx, newInstances)
+			if err != nil {
+				slog.Error("Unable to send data plane status update", "error", err)
```
Comment (olli-holmala) [inline internal/plugin/grpc_client_plugin.go:135]: Is there no `slog.ErrorContext`?
```suggestion
				slog.ErrorContext(ctx, "Unable to send data plane status update", "error", err)
```

File: internal/plugin/process_monitor_plugin.go [hunk 2]
```diff
@@ -105,22 +105,45 @@ func (pm *ProcessMonitor) run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-pm.processTicker.C:
-			newCtx := context.WithValue(ctx, logger.CorrelationIDContextKey, logger.GenerateCorrelationID())
-			slog.DebugContext(newCtx, "Checking for process changes")
+			slog.Debug("Checking for process updates")
```
Comment (olli-holmala) [inline internal/plugin/process_monitor_plugin.go:108]: Why the change to just `slog.Debug()`? Maybe we should clarify when we should log with context and when not to 🤔

Comment (dhurley) [reply internal/plugin/process_monitor_plugin.go:108]: my bad. It should be DebugContext. We should always pass context to the logger if it is available in the function.

**Review by olli-holmala — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

File: internal/plugin/process_monitor_plugin.go [hunk 3]
```diff
@@ -105,22 +105,45 @@ func (pm *ProcessMonitor) run(ctx context.Context) {
...
+				haveChanged = true
+				break
+			}
+		}
+	} else {
+		haveChanged = true
 	}
+
+	return haveChanged
 }
```
Comment (Dean-Coakley) [inline internal/plugin/process_monitor_plugin.go:149]: Totally optional but I think something more like the following would be easier to read:
```go
func haveProcessesChanged(oldProcesses, newProcesses []*model.Process) bool {
	if len(oldProcesses) != len(newProcesses) {
		return true
	}

	oldPIDMap := make(map[int32]struct{})
	for _, oldProcess := range oldProcesses {
		oldPIDMap[oldProcess.Pid] = struct{}{}
	}

	for _, newProcess := range newProcesses {
		if _, ok := oldPIDMap[newProcess.Pid]; !ok {
			return true
		}
	}

	return false
}
```

**Review by Dean-Coakley — APPROVED** [review]

File: internal/plugin/process_monitor_plugin.go (hunk 3 above)
Comment (dhurley) [reply internal/plugin/process_monitor_plugin.go]: Updated and added some comments to make it easier to read.

**Review by craigell — APPROVED** [review]
